wrappers/net/grpc/client_interceptors.go

```diff
@@ -0,0 +1,62 @@
+package epsagongrpc
+
+import (
+	"context"
+	"github.com/epsagon/epsagon-go/epsagon"
+	"github.com/epsagon/epsagon-go/protocol"
+	"github.com/epsagon/epsagon-go/tracer"
+	jsoniter "github.com/json-iterator/go"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"strconv"
+)
+
+const EPSAGON_TRACEID_HEADER_KEY = "epsagon-trace-id"
+
+func InjectEpsagonTracerContextID(ctx context.Context, event *protocol.Event) context.Context {
+	traceID := epsagon.GenerateEpsagonTraceID()
+
+	addTraceIdToEvent(traceID, event)
+
+	return metadata.AppendToOutgoingContext(ctx, EPSAGON_TRACEID_HEADER_KEY, traceID)
+}
+
+func addTraceIdToEvent(traceID string, event *protocol.Event) {
+	event.Resource.Metadata[tracer.EpsagonGRPCTraceIDKey] = traceID
+}
+
+
+// UnaryClientInterceptor returns a new unary server interceptor for OpenTracing.
+func UnaryClientInterceptor(args ...context.Context) grpc.UnaryClientInterceptor {
+
+	wrapperTracer := epsagon.ExtractTracer(args)
+	return func (ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+
+		Event := createGRPCEvent("grpc.client", method, "grpc-client")
```

Comment:
so a lot of calls here can fail and it would be better to add a deffered func which will make sure to invoke the original func in case we have an internal failure
An example is in the fiber handler (so in case of panic we will behave as expected)